### PR TITLE
Add support for Packages API

### LINKED ISSUE #2075: Does not to be any support for the Packages API
I do not see any support for the [Packages API](https://docs.github.com/en/rest/reference/packages). I don't know if I miss it somewhere though.

If it does not exist then I would be happy to work on it.

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
Adds support for [Packages API](https://docs.github.com/en/rest/reference/packages) through expansion of Users and Organization Services.

Would resolve #2075

### CODE REVIEW

File: github/packages.go
Comment (gmlewis) [inline github/packages.go:140]:
```suggestion
	Tags []string `json:"tags,omitempty"`
```

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:29]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:53]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:57]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:41]: Please replace `Passing the empty string will`
with: `Passing the empty string for "user" will`
in this place and all places throughout this PR for consistency
since there are many places where multiple arguments are passed.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:77]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:81]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:82]: Please delete this blank line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:97]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:101]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:102]: Please delete this blank line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:117]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:121]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:127]: Please delete this blank line.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:130]:
```suggestion
// Get a specific version of a package for a user. Passing the empty string for "user" will
```

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:142]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:146]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:152]: Please delete this blank line.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:167]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:171]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:172]: Please delete this blank line.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:187]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:191]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:192]: Please delete this blank line.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:26]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:13]:
```suggestion
// List the packages for an organization.
```

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:43]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:51]:
```suggestion
// Delete a package from an organization.
```

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:73]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:77]:
```suggestion
// Get all versions of a package in an organization.
```

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:86]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:90]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:96]: Please delete this blank line.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:99]:
```suggestion
// Get a specific version of a package in an organization.
```

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:108]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:117]:
```suggestion
// Delete a package version from an organization.
```

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:126]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:127]: Please delete this blank line.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:130]:
```suggestion
// Restore a package version to an organization.
```

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:139]: Please add a blank line above this line for consistency with the rest of the repo.

File: github/orgs_packages.go
Comment (gmlewis) [inline github/orgs_packages.go:140]: Please delete this blank line.

File: github/packages.go
Comment (gmlewis) [inline github/packages.go:121]: Please replace lines 121-125 with the single line:

```
ListOptions
```

File: github/packages.go
Comment (gmlewis) [inline github/packages.go:110]:
```suggestion
// PackageListOptions represents the optional list options for a package.
```

File: github/packages.go
Comment (gmlewis) [inline github/packages.go:116]:
```suggestion
	PackageType *string `url:"package_type,omitempty"`
```

File: github/packages.go
Comment (gmlewis) [inline github/packages.go:128]:
```suggestion
// PackageMetadata represents metadata from a package.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Cyb3r-Jak3 !
This is off to a great start.  I have to run now, but will continue my review later.
This should give you a good idea of what I'm looking for.

**Comment by gmlewis** [comment]
Thank you, @Cyb3r-Jak3 , but I'm going to hold off on a complete review until all unit tests are passing.
It looks like they are currently failing here:

```
    orgs_packages_test.go:208: Request parameters: map[state:[deleted]], want map[page:[2] per_page:[10] state:[deleted]]
--- FAIL: TestOrganizationsService_ListPackagesVersions (0.00s)
```

Please see our CONTRIBUTING.md guide to show you how to run the tests before pushing changes.  Thanks.

**Comment by Cyb3r-Jak3** [comment]
@gmlewis, sorry about that. All good now.

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:21]: Can you please add a unit test for the case when `user` is non-nil?  (`"u"` is fine, and common in other unit tests.)

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:79]: Can you please add a unit test for the case when user is non-nil?

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:100]: Can you please add a unit test for the case when user is non-nil?

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:121]: Can you please add a unit test for the case when user is non-nil?

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:148]: Can you please add a unit test for the case when user is non-nil?

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:175]: Can you please add a unit test for the case when user is non-nil?

File: github/users_packages.go
Comment (gmlewis) [inline github/users_packages.go:196]: Can you please add a unit test for the case when user is non-nil?

File: github/orgs_packages_test.go
Comment (gmlewis) [inline github/orgs_packages_test.go:99]: For every unit test that you added, can you please follow the pattern of other tests in this repo and also call 

```
testBadOptions(t, methodName, func() (err error)
```

?  (Please see `repos_test.go` for many examples.)

This will help with the code coverage.  Thank you!

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/users_packages_test.go
Comment (gmlewis) [inline github/users_packages_test.go:62]:
```suggestion
	})

```

File: github/orgs_packages_test.go
Comment (gmlewis) [inline github/orgs_packages_test.go:58]:
```suggestion
	packages, _, err := client.Organizations.ListPackages(ctx, "o", &PackageListOptions{})
```

File: github/orgs_packages_test.go
Comment (gmlewis) [inline github/orgs_packages_test.go:58]: @Cyb3r-Jak3 - can you please update all the other tests to follow this pattern?

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Review by gmlewis — APPROVED** [review]
Thank you, @Cyb3r-Jak3 !
This looks great!  I really appreciate your patience with me and my reviews.
LGTM.

Awaiting second LGTM before merging.

**Comment by Cyb3r-Jak3** [comment]
@wesleimp Nudge

**Comment by gmlewis** [comment]
@Parker77 - if you have time to do a code review, that would be greatly appreciated.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
> LGTM.

Thank you, @Parker77 !
Merging.

**Pull request merged**
